frame/contracts/src/tests.rs

```diff
@@ -118,10 +118,17 @@ pub struct TestExtension {
 	last_seen_inputs: (u32, u32, u32, u32),
 }
 
+#[derive(Default)]
 pub struct RevertingExtension;
 
+#[derive(Default)]
 pub struct DisabledExtension;
 
+#[derive(Default)]
+pub struct TempStorageExtension {
+	storage: u32,
+}
```

Comment:
So I guess this was something we weren't able to express before, right? How do you see people using this scratch space now?

Reply:
I need it for my XCM chain extension.  There are certain operations which prepare a value and then return it. This is costly when we need to copy it back. So it will be basically:
```Rust
seal_xcm_prepare_execute(xcm: Vec<u8>, dest: MultiLocation) -> Weight;
seal_xcm_execute_prepared() -> Result<()>;
```

The xcm preparation actually returns the prepared xcm. Instead of copying back it will be stored in the temporary storage.